plugins/filetrees/yazi.nix

```diff
@@ -0,0 +1,161 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+
+  inherit (lib.nixvim) defaultNullOpts;
+  types = lib.nixvim.nixvimTypes;
+in
+nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "yazi";
+  originalName = "yazi.nvim";
+  defaultPackage = pkgs.vimPlugins.yazi-nvim;
+
+  maintainers = [ maintainers.khaneliman ];
+
+  description = ''
+    Yazi is a blazing fast file manager for the terminal.
+    This plugin allows you to open yazi in a floating window in Neovim.
+
+    The default grep functionality requires `telescope`.
+    You can override the grep functionality through the `plugins.yazi.settings.integrations.grep_in_directory`
+    and `plugins.yazi.settings.integrations.grep_in_selected_files` if you do not wish to
+    include `telescope` as an extra dependency.
```

Comment:
```suggestion
    You can override the grep functionality through the `plugins.yazi.settings.integrations.grep_in_directory`
    and `plugins.yazi.settings.integrations.grep_in_selected_files` options
    if you do not wish to use `telescope`.
    Otherwise you must enable `plugins.telescope` or install it some other way.
```